fastf1/core.py

```diff
@@ -1909,6 +1910,63 @@ def _calculate_quali_like_session_results(self, force=False):
         self.results.loc[:, quali_results.columns] = quali_results
         self.results.sort_values(by=['Position'], inplace=True)
 
+    @soft_exceptions(
+        "race results",
+        "Failed to calculate race results from lap times!",
+        _logger,
+    )
+    def _calculate_race_like_session_results(self, force=False):
+        """
+        Try to calculate race results from lap times if no results are
+        available.
+
+        DNF drivers are ordered based on the number of laps they completed.
+        Do not rely on this data as FIA considers these drivers to have no
+        finishing positions.
+
+        Args:
+            force (bool): Force calculation of race results even if
+            results are already available, (default: False)
+        """
+
+        if self.name not in self._RACE_LIKE_SESSIONS:
+            return
+
+        if not hasattr(self, '_laps'):
+            return
+
+        if not self.results['Position'].isna().all() and not force:
+            # Don't do anything if results are already available
+            # unless force is True
+            return
+
+        if self.laps['Deleted'].dtype.name != 'bool':
+            _logger.warning(
+                "Cannot calculate race results: missing information "
+                "about deleted laps. Make sure that race control messages are "
+                "being loaded."
+            )
+
+        # This keeps the final lap of each driver
+        final_laps = self.laps.drop_duplicates(
+            subset="DriverNumber", keep="last"
+        )
```

Comment:
I'd prefer not to rely on the input order here, which this seems to do. What about the following instead? (not fully tested)
```suggestion
        final_laps = (self.laps.groupby("DriverNumber")
                      .agg({"LapNumber": "max"}))
```

Reply:
Fair enough. I believe your change would only keep the `LapNumber` column but I will implement similar logic